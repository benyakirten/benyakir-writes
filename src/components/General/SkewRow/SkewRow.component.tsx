import * as React from "react";
import { AntiSkewed, Skewed } from "./SkewRow.styles";

const SkewRow: React.FC<ChildrenProp> = ({ children }) => {
	return (
		<Skewed>
			<AntiSkewed>{children}</AntiSkewed>
		</Skewed>
	);
};

export default SkewRow;
