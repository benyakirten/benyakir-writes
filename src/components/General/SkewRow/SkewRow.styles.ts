import styled from 'styled-components'

import { media } from '@Styles/queries'
import { SHADOW_LG } from '@StyleVars'
import { convertHexToRGBA } from '@Utils/colors'

export const Skewed = styled.section<{ width?: string }>`
  display: flex;
  justify-content: space-between;
  align-items: center;

  margin: 0 2rem;
  padding: 2rem 3rem;

  box-shadow: ${(props) =>
    `${SHADOW_LG} ${convertHexToRGBA(props.theme.base.shadowColor, 0.4)}`};
  border: 2px solid ${(props) => props.theme.base.border};
  background-color: ${(props) => props.theme.skewRow.background};

  transform: skew(-25deg);

  ${media.phone} {
    transform: skew(0deg);
    min-width: 20rem;
    margin: 0;
  }
`

export const AntiSkewed = styled.div`
  display: flex;
  justify-content: space-around;

  ${media.phone} {
    flex-direction: column;
  }

  width: 100%;

  transform: skew(25deg);

  ${media.phone} {
    transform: skew(0deg);
  }

  ${media.reducedMotion} {
    transform: skew(0deg);
  }
`
