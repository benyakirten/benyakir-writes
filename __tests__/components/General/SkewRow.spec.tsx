import * as React from "react";

import { render, cleanup } from "@TestUtils";
import SkewRow from "@Gen/SkewRow/SkewRow.component"

describe('SkewRow component', () => {
    afterEach(cleanup)

    it('should render properly', () => {
        expect(() => render(<SkewRow />)).not.toThrow()
    })
})