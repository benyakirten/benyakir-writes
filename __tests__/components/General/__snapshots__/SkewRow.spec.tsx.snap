// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkewRow component should render properly 1`] = `
<section
  className="sc-bdvvtL krqzEm"
>
  <div
    className="sc-gsDKAQ MhxVF"
  />
</section>
`;
